swarm/src/lib.rs

```diff
@@ -502,19 +503,13 @@ where
     /// swarm.dial("/ip6/::1/tcp/12345".parse::<Multiaddr>().unwrap());
     /// ```
     pub fn dial(&mut self, opts: impl Into<DialOpts>) -> Result<(), DialError> {
```

Comment:
I opted not to return the `ConnectionId` here even though it might at first be counter-intuitive. The user already has access to it via `DialOpts` so they can get it if needed but it is a little bit more verbose.

Reply:
If we return the ID, it gives the impression that we are creating it, which isn't true.

Reply:
No strong opinion. Fine with either. Let's keep as is.